Slurm workslow submits intial jobs but then hangs until ctrl-c

<!-- Please do not post usage questions here. Ask them on Stack Overflow: https://stackoverflow.com/questions/tagged/snakemake -->

**Software Versions**

$ snakemake --version
8.23.0
$ mamba list | grep "snakemake-executor-plugin-slurm"
$ conda list | grep "snakemake-executor-plugin-slurm"
snakemake-executor-plugin-slurm 0.11.0             pyhdfd78af_0    bioconda
snakemake-executor-plugin-slurm-jobstep 0.2.1              pyhdfd78af_0    bioconda
$ sinfo --version
slurm 24.05.3

**Describe the bug**
When starting a snakemake workflow on a slurm cluster (surf/sara Snellius), the workflow starts but hangs on the initially submitted jobs (that seem to complete and disappear from the squeue overview). It's like Snakemake does not get the signal that the jobs finish. This is my config:

```yaml
executor: slurm
default-resources:
  slurm_partition: "rome"
  time: 1h
  # slurm_extra: "'-o cluster_outputs/smk.{rule}.{jobid}.out -e cluster_outputs/smk.{rule}.{jobid}.err'"
printshellcmds: True
jobs: 100
restart-times: 3
latency-wait: 60
rerun-incomplete: True
use-conda: True
conda-prefix: /home/me/projects/snaqs_files/snakemake_envs

set-threads:
    salmon: 16

set-resources:
    # rule specific resources
  fastqc:
    slurm_partition: staging
```


**Logs**
This is how it ends (I pressed ctrl-c at this line `^CTerminating processes on user request, this might take some time.`):

```bash
[Tue Oct 15 18:56:16 2024]
rule fastqc:
    input: fastq_raw/0053_P2017BB3S19R_S1_R1_001.fastq.gz
    output: qc/fastqc/0053_P2017BB3S19R_S1_R1_001_fastqc.html, qc/fastqc/0053_P2017BB3S19R_S1_R1_001_fastqc.zip
    jobid: 16
    benchmark: benchmark/0053_P2017BB3S19R_S1_R1.fastqc.fastqc.benchmark.tsv
    reason: Missing output files: qc/fastqc/0053_P2017BB3S19R_S1_R1_001_fastqc.html, qc/fastqc/0053_P2017BB3S19R_S1_R1_001_fastqc.zip
    wildcards: sample=0053_P2017BB3S19R_S1, read=R1
    resources: mem_mb=1000, mem_mib=954, disk_mb=1000, disk_mib=954, tmpdir=<TBD>, slurm_partition=staging, time=1h


        fastqc fastq_raw/0053_P2017BB3S19R_S1_R1_001.fastq.gz --outdir=qc/fastqc
        
No wall time information given. This might or might not work on your cluster. If not, specify the resource runtime in your rule or as a reasonable default via --default-resources.
Job 16 has been submitted with SLURM jobid sbatch: Single-node jobs run on a shared node by default. Add --exclusive if you want to use a node exclusively.
sbatch: A full node consists of 32 CPU cores, 229376 MiB of memory and 0 GPUs and can be shared by up to 32 jobs.
sbatch: By default shared jobs get 7168 MiB of memory per CPU core, unless explicitly overridden with --mem-per-cpu, --mem-per-gpu or --mem.
sbatch: You will be charged for 1 CPUs, based on the number of CPUs and the amount memory that you've requested.
8178969 (log: /gpfs/home5/me/temp/test_pipeline/.snakemake/slurm_logs/rule_fastqc/0053_P2017BB3S19R_S1_R1/sbatch: Single-node jobs run on a shared node by default. Add --exclusive if you want to use a node exclusively.
sbatch: A full node consists of 32 CPU cores, 229376 MiB of memory and 0 GPUs and can be shared by up to 32 jobs.
sbatch: By default shared jobs get 7168 MiB of memory per CPU core, unless explicitly overridden with --mem-per-cpu, --mem-per-gpu or --mem.
sbatch: You will be charged for 1 CPUs, based on the number of CPUs and the amount memory that you've requested.
8178969.log).
^CTerminating processes on user request, this might take some time.
WorkflowError:
Unable to cancel jobs with scancel (exit code 127): scancel: unrecognized option '--exclusive'
Try "scancel --help" for more information
/bin/sh: line 2: sbatch:: command not found
/bin/sh: line 3: sbatch:: command not found
/bin/sh: line 4: sbatch:: command not found
/bin/sh: line 9: 8178965: command not found
/bin/sh: line 10: sbatch:: command not found
/bin/sh: line 11: sbatch:: command not found
/bin/sh: line 12: sbatch:: command not found
/bin/sh: line 17: 8178967: command not found
/bin/sh: line 18: sbatch:: command not found
/bin/sh: line 19: sbatch:: command not found
/bin/sh: line 20: sbatch:: command not found
/bin/sh: line 25: 8178969: command not found

```

**Minimal example**
This is a big effort, if it's really required, I'll try to make a minimal pipelines. Apologies.

**Additional context**
I have also posted a question on [https://bioinformatics.stackexchange.com/questions/22963/snakemake-on-a-slurm-cluster](bioinformatics.stackexchange), however there I ask for support for the generic executor as well (which doesn't let met specify the partition).

I think in general, some more extensive examples of profiles where all options are used would be nice to have. And, perhaps Snellius is deviating from normal Slurm?

Hi @cmeesters I tried using poetry, using this pyproject.toml:

```toml
[tool.poetry]
name = "snaqs"
version = "2.0.0"
description = ""
authors = ["Freek... <freek@x.x>"]
readme = "README.md"
packages = []
# package-mode = false

[tool.poetry.dependencies]
python = "^3.11"
pandas = "*"
snakemake = "=8.25.0"
snakemake-executor-plugin-slurm = {git = "git@github.com:snakemake/snakemake-executor-plugin-slurm.git", branch='fix/sbatch-stderr-parsing'}

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

It fails with this error:

```bash
(base) [fvhemert@int6 snaqs]$ poetry install
The currently activated Python version 3.10.14 is not supported by the project (^3.11).
Trying to find and use a compatible version. 
Using python3.11 (3.11.7)
Creating virtualenv snaqs-Oei8VcMh-py3.11 in /home/fvhemert/.cache/pypoetry/virtualenvs
Updating dependencies
Resolving dependencies... (64.8s)
Resolving dependencies... (77.9s)

Package operations: 54 installs, 0 updates, 0 removals

  - Installing attrs (24.2.0)
  - Installing rpds-py (0.20.1)
  - Installing argparse-dataclass (2.0.0)
  - Installing configargparse (1.7)
  - Installing referencing (0.35.1)
  - Installing jsonschema-specifications (2024.10.1)
  - Installing platformdirs (4.3.6)
  - Installing smmap (5.0.1)
  - Installing snakemake-interface-common (1.17.4)
  - Installing throttler (1.2.2)
  - Installing traitlets (5.14.3)
  - Installing certifi (2024.8.30)
  - Installing charset-normalizer (3.4.0)
  - Installing dpath (2.2.0)
  - Installing fastjsonschema (2.20.0)
  - Installing gitdb (4.0.11)
  - Installing idna (3.10)
  - Installing jsonschema (4.23.0)
  - Installing jupyter-core (5.7.2)
  - Installing markupsafe (3.0.2)
  - Installing plac (1.4.3)
  - Installing pyyaml (6.0.2)
  - Installing reretry (0.11.8)
  - Installing six (1.16.0)
  - Installing snakemake-interface-executor-plugins (9.3.2)
  - Installing urllib3 (2.2.3)
  - Installing wrapt (1.16.0)
  - Installing appdirs (1.4.4)
  - Installing conda-inject (1.3.2)
  - Installing connection-pool (0.0.3)
  - Installing datrie (0.8.2): Failed

  ChefBuildError

  Backend subprocess exited when trying to invoke build_wheel
  
  /scratch-local/70716/tmp4nt0tivw/.venv/lib/python3.11/site-packages/setuptools/_distutils/dist.py:261: UserWarning: Unknown distribution option: 'tests_require'
    warnings.warn(msg)
  running bdist_wheel
  running build
  running build_clib
  building 'datrie' library
  creating build/temp.linux-x86_64-cpython-311/libdatrie/datrie
  gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Ilibdatrie -c libdatrie/datrie/alpha-map.c -o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/alpha-map.o
  libdatrie/datrie/alpha-map.c: In function ‘alpha_map_char_to_trie’:
  libdatrie/datrie/alpha-map.c:500:21: warning: comparison of integer expressions of different signedness: ‘TrieIndex’ {aka ‘int’} and ‘AlphaChar’ {aka ‘unsigned int’} [-Wsign-compare]
    500 |     if (alpha_begin <= ac && ac <= alpha_map->alpha_end)
        |                     ^~
  gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Ilibdatrie -c libdatrie/datrie/darray.c -o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/darray.o
  libdatrie/datrie/darray.c: In function ‘da_fread’:
  libdatrie/datrie/darray.c:239:22: warning: comparison of integer expressions of different signedness: ‘TrieIndex’ {aka ‘int’} and ‘long unsigned int’ [-Wsign-compare]
    239 |     if (d->num_cells > SIZE_MAX / sizeof (DACell))
        |                      ^
  gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Ilibdatrie -c libdatrie/datrie/dstring.c -o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/dstring.o
  gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Ilibdatrie -c libdatrie/datrie/fileutils.c -o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/fileutils.o
  libdatrie/datrie/fileutils.c: In function ‘file_read_chars’:
  libdatrie/datrie/fileutils.c:103:52: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
    103 |     return (fread (buff, sizeof (char), len, file) == len);
        |                                                    ^~
  libdatrie/datrie/fileutils.c: In function ‘file_write_chars’:
  libdatrie/datrie/fileutils.c:109:53: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
    109 |     return (fwrite (buff, sizeof (char), len, file) == len);
        |                                                     ^~
  gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Ilibdatrie -c libdatrie/datrie/tail.c -o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/tail.o
  libdatrie/datrie/tail.c: In function ‘tail_fread’:
  libdatrie/datrie/tail.c:144:22: warning: comparison of integer expressions of different signedness: ‘TrieIndex’ {aka ‘int’} and ‘long unsigned int’ [-Wsign-compare]
    144 |     if (t->num_tails > SIZE_MAX / sizeof (TailBlock))
        |                      ^
  gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Ilibdatrie -c libdatrie/datrie/trie-string.c -o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/trie-string.o
  gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Ilibdatrie -c libdatrie/datrie/trie.c -o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/trie.o
  ar rcs build/temp.linux-x86_64-cpython-311/libdatrie.a build/temp.linux-x86_64-cpython-311/libdatrie/datrie/alpha-map.o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/darray.o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/dstring.o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/fileutils.o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/tail.o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/trie-string.o build/temp.linux-x86_64-cpython-311/libdatrie/datrie/trie.o
  running build_ext
  building 'datrie' extension
  creating build/temp.linux-x86_64-cpython-311/src
  gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Ilibdatrie -I/scratch-local/70716/tmp4nt0tivw/.venv/include -I/usr/include/python3.11 -c src/datrie.c -o build/temp.linux-x86_64-cpython-311/src/datrie.o
  src/datrie.c:36:10: fatal error: Python.h: No such file or directory
     36 | #include "Python.h"
        |          ^~~~~~~~~~
  compilation terminated.
  error: command '/usr/bin/gcc' failed with exit code 1
  

  at /gpfs/home5/fvhemert/.local/share/pypoetry/venv/lib/python3.10/site-packages/poetry/installation/chef.py:164 in _prepare
      160│ 
      161│                 error = ChefBuildError("\n\n".join(message_parts))
      162│ 
      163│             if error is not None:
    → 164│                 raise error from None
      165│ 
      166│             return path
      167│ 
      168│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:

Note: This error originates from the build backend, and is likely not a problem with poetry but with datrie (0.8.2) not supporting PEP 517 builds. You can verify this by running 'pip wheel --no-cache-dir --use-pep517 "datrie (==0.8.2)"'.

  - Installing docutils (0.21.2)
  - Installing gitpython (3.1.43)
  - Installing humanfriendly (10.0)
  - Installing immutables (0.21)
  - Installing jinja2 (3.1.4)
  - Installing nbformat (5.10.4)
  - Installing numpy (2.1.2)
  - Installing packaging (24.1)
  - Installing psutil (6.1.0)
  - Installing pulp (2.9.0)
  - Installing python-dateutil (2.9.0.post0)
  - Installing pytz (2024.2)
  - Installing requests (2.32.3)
  - Installing smart-open (7.0.5)
  - Installing snakemake-executor-plugin-slurm-jobstep (0.2.1)
  - Installing snakemake-interface-report-plugins (1.1.0)
  - Installing snakemake-interface-storage-plugins (3.3.0)
  - Installing tabulate (0.9.0)
  - Installing tzdata (2024.2)
  - Installing yte (1.5.4)
```

--- a/snakemake_executor_plugin_slurm/__init__.py
+++ b/snakemake_executor_plugin_slurm/__init__.py
@@ -244,6 +244,8 @@ class Executor(RemoteExecutor):
         # To extract the job id we split by semicolon and take the first element
         # (this also works if no cluster name was provided)
         slurm_jobid = out.strip().split(";")[0]
+        if not slurm_jobid:
+            raise WorkflowError("Failed to retrieve SLURM job ID from sbatch output.")
         slurm_logfile = slurm_logfile.replace("%j", slurm_jobid)
         self.logger.info(
             f"Job {job.jobid} has been submitted with SLURM jobid {slurm_jobid} "

--- a/snakemake_executor_plugin_slurm/__init__.py
+++ b/snakemake_executor_plugin_slurm/__init__.py
@@ -217,7 +217,7 @@ class Executor(RemoteExecutor):
 
         self.logger.debug(f"sbatch call: {call}")
         try:
-            process = subprocess.Popen(call, hell=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            process = subprocess.Popen(call, shell=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             out, err = process.communicate()
             if process.returncode != 0:
                 raise subprocess.CalledProcessError(

--- a/snakemake_executor_plugin_slurm/__init__.py
+++ b/snakemake_executor_plugin_slurm/__init__.py
@@ -217,7 +217,13 @@ class Executor(RemoteExecutor):
 
         self.logger.debug(f"sbatch call: {call}")
         try:
-            process = subprocess.Popen(call, shell=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            process = subprocess.Popen(
+                call,
+                shell=True,
+                text=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
             out, err = process.communicate()
             if process.returncode != 0:
                 raise subprocess.CalledProcessError(
@@ -227,7 +233,7 @@ class Executor(RemoteExecutor):
             raise WorkflowError(
                 f"SLURM job submission failed. The error message was {e.output}"
             )
-        if err: # any other error message?
+        if err:  # any other error message?
             raise WorkflowError(
                 f"SLURM job submission failed. The error message was {err}"
             )

--- a/snakemake_executor_plugin_slurm/__init__.py
+++ b/snakemake_executor_plugin_slurm/__init__.py
@@ -217,12 +217,19 @@ class Executor(RemoteExecutor):
 
         self.logger.debug(f"sbatch call: {call}")
         try:
-            out = subprocess.check_output(
-                call, shell=True, text=True, stderr=subprocess.STDOUT
-            ).strip()
+            process = subprocess.Popen(call, hell=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            out, err = process.communicate()
+            if process.returncode != 0:
+                raise subprocess.CalledProcessError(
+                    process.returncode, call, output=err
+                )
         except subprocess.CalledProcessError as e:
             raise WorkflowError(
                 f"SLURM job submission failed. The error message was {e.output}"
+            )
+        if err: # any other error message?
+            raise WorkflowError(
+                f"SLURM job submission failed. The error message was {err}"
             )
 
         # multicluster submissions yield submission infos like
@@ -230,7 +237,7 @@ class Executor(RemoteExecutor):
         # --parsable option it simply yields "<id>;<name>".
         # To extract the job id we split by semicolon and take the first element
         # (this also works if no cluster name was provided)
-        slurm_jobid = out.split(";")[0]
+        slurm_jobid = out.strip().split(";")[0]
         slurm_logfile = slurm_logfile.replace("%j", slurm_jobid)
         self.logger.info(
             f"Job {job.jobid} has been submitted with SLURM jobid {slurm_jobid} "

